core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -359,4 +435,69 @@ class SpendReportingService(
         case ex: Exception =>
           Future.failed(RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, ex)))
       }
+
+//  def getSpendForUserWorkspaces(
+//    start: DateTime,
+//    end: DateTime
+//  ): Future[SpendReportingResults] =
+//    for {
+//      workspaces <- getOwnerWorkspaces()
+//      billing <- getBillingForWorkspaces(workspaces)
+//      query = getAllUserWorkspaceQuery(billing)
+//
+//    } yield query
+
+  def getSpendForAllWorkspaces(
+    project: RawlsBillingProjectName,
+    start: DateTime,
+    end: DateTime,
+    aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] = {
+    validateReportParameters(start, end)
+    requireProjectAction(project, SamBillingProjectActions.readSpendReport) {
+      for {
+        spendExportConf <- getSpendExportConfiguration(project)
+        projectNames <- getWorkspaceGoogleProjects(project)
+
+        query = getQuery(aggregations, spendExportConf)
+        queryJob = setUpQuery(query, spendExportConf, start, end, projectNames)
+
+        job: Job <- bigQueryService.use(_.runJob(queryJob)).unsafeToFuture().map(_.waitFor())
+        _ = logSpendQueryStats(job.getStatistics[JobStatistics.QueryStatistics])
+        result = job.getQueryResults()
+      } yield result.getValues.asScala.toList match {
+        case Nil =>
+          throw RawlsExceptionWithErrorReport(
+            StatusCodes.NotFound,
+            s"no spend data found for billing project ${project.value} between dates ${toISODateString(start)} and ${toISODateString(end)}"
+          )
+        case rows => extractSpendReportingResults(rows, start, end, projectNames, aggregations)
+      }
+    }
+  }
+
+  def getOwnerWorkspaces(): Future[Seq[WorkspaceListResponse]] =
+    workspaceServiceConstructor(ctx).listWorkspaces(WorkspaceFieldSpecs(), -1) map {
+      case JsArray(jsArray) =>
+        val workspaces = jsArray.map(_.convertTo[WorkspaceListResponse])
+        workspaces.filter(_.accessLevel == Owner)
+      case _ => throw new IllegalArgumentException("Expected a JsArray")
+    }
+
+  def getBillingForWorkspaces(
+    workspaces: Seq[WorkspaceListResponse]
+  ): Future[Map[BillingProjectSpendExport, Seq[GoogleProjectId]]] = {
+    val groupedWorkspaces = workspaces.groupBy(_.workspace.namespace)
+    val billingFutures: Iterable[Future[Option[(BillingProjectSpendExport, Seq[GoogleProjectId])]]] =
+      groupedWorkspaces.map { case (namespace, wsList) =>
+        getSpendExportConfiguration(RawlsBillingProjectName(namespace)).map { exportConfig =>
+          Some(
+            exportConfig -> wsList
+              .map(_.workspace.googleProject)
+          )
+        }
+      }
+    Future.sequence(billingFutures).map(_.flatten.toMap)
+  }
```

Comment:
I also see this method only called from a unit test. Is it used? If so: this method loops over each namespace and gets its spend config via `getSpendExportConfiguration`. Each invocation of this opens a new db transaction, queries for the config for a single namespace, and completes that transaction.

Could you instead write a new method that accepts a list of namespaces and makes one SQL query for all those namespaces and returns their result?

Reply:
Same answer re: it being used.  I will try to rewrite.